### PR TITLE
Implement dds_instance_v3 resource

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
Add new resource `opentelekomcloud_dds_instance_v3`

## PR Checklist

* [x] Refers to: #619
* [x] Tests added/passed.
* [x] Documentation added.
* [ ] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccDDSV3Instance_basic
--- PASS: TestAccDDSV3Instance_basic (831.62s)
=== RUN   TestAccDDSV3Instance_minConfig
--- PASS: TestAccDDSV3Instance_minConfig (797.40s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: docs/resources/dds_instance_v3.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  storage space is 20 GB. This parameter is invalid for `mongos` nodes. Therefore, you do not need
+  to specify the storage space for `mongos` nodes.
+  * For a `replica set` instance, the value ranges from 10 to 2000.
+-> **Note:** This parameter is mandatory for all nodes except `mongos`. This parameter is invalid
+  for the `mongos` nodes.
+
+* `spec_code` - (Required) Specifies the resource specification code. In a cluster instance,
+  multiple specifications need to be specified. All specifications must be of the same series,
+  that is, general-purpose (s6), enhanced (c3), or enhanced II (c6). For example:
+  * dds.mongodb.s6.large.4.mongos and dds.mongodb.s6.large.4.config have the same specifications.
```
Comment (outcatcher) [inline docs/resources/dds_instance_v3.md]: Please use `code formatting` for flavors, e.g. `dds.mongodb.s6.large.4.mongos`

File: opentelekomcloud/provider.go [hunk 2]
```diff
@@ -372,6 +372,7 @@ func Provider() terraform.ResourceProvider {
 			"opentelekomcloud_vpnaas_site_connection_v2":          resourceVpnSiteConnectionV2(),
 			"opentelekomcloud_logtank_group_v2":                   resourceLTSGroupV2(),
 			"opentelekomcloud_logtank_topic_v2":                   resourceLTSTopicV2(),
+			"opentelekomcloud_dds_instance_v3":                    resourceDdsInstanceV3(),
```
Comment (outcatcher) [inline opentelekomcloud/provider.go]: Please move it up to be as alphabetically sorted as possible

File: opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+			},
+			"backup_strategy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_time": {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go:157]: Please add validation if possible:
```
    The HH value must be 1 greater than the hh value.
    The values of mm and MM must be the same and must be set to any of the following: 00, 15, 30, or 45.
```

File: opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"keep_days": {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go:162]: Please add validation: `The value range is from 0 to 732.`

File: opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+							ValidateFunc: validation.StringInSlice([]string{
+								"DDS-Community",
+							}, true),
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"3.4", "3.2",
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go]: It doesn't seem to be a good idea to validate version range, as version tend to change between OTC releases

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**